### DISCUSSION TITLE
How to use "public" directory with Bun Build?

### DISCUSSION THREAD

**Discussion by phenomen** [oven-sh/bun#8561] [Q&A]
Currently, I use Vite to build my library and looking forward to switching to Bun.

In Vite I have a `src` directory with code and assets needs to be bundled/minified and also `public` directory for static assets: css, images, fonts and some json files that should not be transformed in any way and copied into build as-is.

But after glancing at Bun docs, I cannot find any equivalent of `publicDir` config parameter. How can I handle static assets with Bun.Build?

Vite docs reference: https://vitejs.dev/config/shared-options.html#publicdir

**Comment by ErsinCabuk** [comment]
I'm having the same problem. Have you found anything?

**Comment by mirogta** [comment]
You can combine bun build with copying the `public` directory to the output directory. I've observed that `bun` also doesn't automatically clean up the output directory, so I'm running `clean` at the beginning. This is how it looks in the scripts in the `package.json`:
```
  "scripts": {
    "clean": "rm -rf dist",
    "dev": "bun run clean && bun build index.html --outdir dist --watch & nodemon --watch public --exec \"cp -r public/. dist/\"",
    "build": "bun run clean && bun build index.html --outdir dist --production && cp -r public/. dist/",
    "preview": "bun run dist/index.js"
  },
```

**Comment by SX110903** [comment]
bun build no tiene un equivalente de public directory como vite, pero puedes copiarlo manualmente en el script de build. en package.json algo asi:

```json
{
  "scripts": {
    "build": "bun build ./src/index.ts --outdir ./dist && cp -r ./public ./dist/public"
  }
}
```

o si quieres algo mas elaborado puedes usar el plugin system de bun para copiar archivos al final del build. pero la verdad para la mayoria de casos el cp en el script es lo mas simple y funciona igual de bien. si estas en windows cambia cp -r por xcopy o usa un script en js que copie la carpeta
